### ISSUE TITLE
On/Off toggle button on instruments

### ISSUE THREAD

**Issue by musikBear** [LMMS/lmms#2685]
I suggest a small on/off toggle button directly on the instrument UI
![onofftogler](https://cloud.githubusercontent.com/assets/6662066/13877722/ed83a3e6-ed0c-11e5-9fdc-f44422a7bbcf.png)
That would make it possible to _activate_ the instrument on the instrument UI and not only on song-editors  _mute/unmute_ diode. The state of the new diode and the song-editor _mute/unmute_ diode, should of cause be synchronized.

**Comment by tresf** [comment]
I think what you're asking for is to bring the mute and solo buttons over to the instrument screen.  Real-estate is probably the biggest issue currently, but the idea has merit.

Is this a piece of cabbage? :laughing: 
![image](https://cloud.githubusercontent.com/assets/6345473/13878200/25d487be-ece6-11e5-9a54-0fa21ec634a0.png)

**Comment by musikBear** [comment]
> Is this a piece of cabbage?

:japanese_ogre: Mean man you - thats a diode :gem: 
Its just to be able to un-mute an instrument/ preset, without having to click in song-editor, and cause the instrument to 'hide'
Could even actually be interesting in piano-roll, but there the handle situation would be even worse.

**Comment by tresf** [comment]
> Its just to be able to un-mute an instrument/ preset, without having to click in song-editor, and cause the instrument to 'hide'

FYI, as a stop-gap, you can use "Always on top" with any widget. :+1: 

Not the same, but helpful on occasion...

![image](https://cloud.githubusercontent.com/assets/6345473/13879635/d9e65900-ecee-11e5-9225-5cd5340948ae.png)

**Comment by musikBear** [comment]
> FYI, as a stop-gap, you can use "Stay on top" with any widget Not the same, but helpful on occasion...

Yes i know that, thanks. I even suggested this as default behavior in 2087 (not dir. related)

> (I strongly favor a better behavior for 'smaller windows', in order to bring them forward, and have them staying there. Imo that could be default behavior, but a small 'tag-icon' in the window tittle-bar, could work. It is to cumbersome to use the context-menu for every window that opens (imo:) )

 It is possible to set for *instruments* in a template, because windows attributes are preserved, even when presets are exchanged! But this is not possible for all effect-inserts, they have to be changed individually as they are loaded. Stay on top as default behavior for all non main components, would make sense, imo.

**Comment by rubiefawn** [comment]
> I think what you're asking for is to bring the mute and solo buttons over to the instrument screen. Real-estate is probably the biggest issue currently, but the idea has merit.

I should be able to do this. They should probably be to the left of the volume knob, and the other stuff might be able to be moved slightly left to make room. Does that sound okay?

**Comment by musikBear** [comment]
> I should be able to do this. They should probably be to the left of the volume knob, and the other stuff might be able to be moved slightly left to make room. Does that sound okay?

Anywhere where you can place a 'diode' it could be any shape. The one UI element that kind of 'stand out' as odd is the save-button, maybe some redesign could create a bit of realestate

**Comment by rubiefawn** [comment]
The mute and solo buttons are rather small, so it fits right in. The only potential problem I see here is that the volume and pan controls are now next to the pitch bend controls.

![Image](https://github.com/user-attachments/assets/562eb189-9c34-4b47-b193-3ee21862080d)

**Comment by musikBear** [comment]
Well done 
> The only potential problem I see here is that the volume and pan controls are now next to the pitch bend controls.

That is insignificant imo.

**Comment by tresf** [comment]
This feature is now available on master branch, available with nightly download once the CI jobs for 6233c5b are completed.
